image_to_json.py

```diff
@@ -1,23 +1,65 @@
 #!/usr/bin/env python
-"""
-Opens image and converts to csv data using the PIL library
+"""Converts image to json-like file.
+
+Usage:
+    image_to_json.py (-i FILE) [-o FILE] [-s K] [-f FILTER]
+
+Options:
+    -h --help                   Shows this screen.
+    -i FILE --input=FILE        Input image file.
+    -o FILE --output=FILE       Output json file. [default: image.json]
+    -s K --size=K               Resize image in K. [default: False]
+    -f FILTER --filter=FILTER   Resampling filter, NEAREST, BILINEAR, BICUBIC or LANCZOS. [default: NEAREST]
 """
 
-import Image
-from sys import argv, exit
+
+import os
+import sys
+import getopt
+from docopt import docopt
+
 import json
 
-if len(argv) != 2:
-    print "USAGE: python image_to_csv.py *image_name*"
-    exit()
+from PIL import Image
+
+
+def get_filter(resample):
+    if resample.upper() == 'NEAREST':
+        f = Image.NEAREST
+    elif resample.upper() == 'BILINEAR':
+        f = Image.BILINEAR
+    elif resample.upper() == 'BICUBIC':
+        f = Image.BICUBIC
+    elif resample.upper() == 'LANCZOS':
+        f = Image.LANCZOS
+    else:
+        f = Image.NEAREST
+    return f
+
+def get_pixels(fname, size=False, resample='NEAREST'):
+    image = Image.open(fname)
+    width, height = image.size
+
+    if size.upper() != 'FALSE':
+        width = int(width*float(size))
+        height = int(height*float(size))
+        image = image.resize((width, height), resample=get_filter(resample))
```

Comment:
You can replace `get_filter(resample)` with `getattr(Image, resample.upper())`. This lets you delete `get_filter` altogether.